### PR TITLE
Split package dependencies

### LINKED ISSUE #362: Reduce mandatory packages bloat
I installed the package in a new machine and it installed the following seldom-used packages:

`pymeeus, convertdate, korean-lunar-calendar`

Any work to reduce (and/or standardize) the number of packaged installed would be helpful to combat package bloat.  For example, I have no need for Korean holidays yet I find myself having to install a package that, by its name, is required only for those type of holidays.

### PR DESCRIPTION
**Pull request by arkid15r**
Resolves #362 

Redesign extras_require logic.
  - Use `importlib` to determine specific package availability
  - Update README.rst
  - Use `mock` to recover 100% coverage
  - Add `pymeeus` option for Chile and Japan

Based on #729 
> Split out three dependencies that are only used for some specific
> countries so they're not installed by default anymore. This is a
> breaking change for consumers using one or more of the affected
> countries. The advantage is that consumers who only use one or a few
> simple calendars don't need to install dependencies that will be unused.

### CODE REVIEW

**Comment by stianjensen** [comment]
Great that you're continuing this and found a way to test it! :+1:

**Review by stianjensen — COMMENTED** [review]
This looks good to me now – anything I can do to help this get merged?

**Comment by stianjensen** [comment]
I guess this will be a breaking change, since it means that existing users of any package that needs an 'extra', will need to update their dependency to add it. So we will need a 1.0 or similar (depending on versioning schemes, any version before 1.0 is technnically breaking).

**Comment by arkid15r** [comment]
Hi @stianjensen,
thanks for checking on this :)

Yeah, at this point the PR is pretty much ready and just sitting in the queue (for a while I have to admit). 
My plan is to merge it as soon as possible after I'm done with #865. The optimistic ETA for #889 release is early April.

I appreciate your help offering!

File: README.rst [hunk 1]
```diff
@@ -113,376 +113,498 @@ for a subdivision its `ISO 3166-2 code`_. The following countries and
...
      - None
+     - None
    * - India
      - IN
      - States: AN, AP, AR, AS, BR, CG, CH, DD, DH, DL, GA, GJ, HP, HR, JH, JK, KA, KL, LA, LD, MH, ML, MN, MP, MZ, NL, OR, PB, PY, RJ, SK, TN, TR, TS, UK, UP, WB
+     - None
    * - Indonesia
      - ID
      - None
+     - None
```
Comment (KJhellico) [inline README.rst:319]:
```suggestion
     - holidays[islamic]
```

File: README.rst [hunk 2]
```diff
@@ -113,376 +113,498 @@ for a subdivision its `ISO 3166-2 code`_. The following countries and
...
+     - None
    * - Norway
      - NO
      - None
+     - None
    * - Pakistan
      - PK
      - None
+     - None
```
Comment (KJhellico) [inline README.rst:455]:
```suggestion
     - holidays[islamic]
```

File: README.rst [hunk 3]
```diff
@@ -113,376 +113,498 @@ for a subdivision its `ISO 3166-2 code`_. The following countries and
...
    * - San Marino
      - SM
      - None
+     - None
    * - Saudi Arabia
      - SA
      - None
+     - None
```
Comment (KJhellico) [inline README.rst:499]:
```suggestion
     - holidays[islamic]
```

File: README.rst [hunk 4]
```diff
@@ -113,376 +113,498 @@ for a subdivision its `ISO 3166-2 code`_. The following countries and
...
      - Cantons: AG, AR, AI, BL, BS, BE, FR, GE, GL, GR, JU, LU, NE, NW, OW, SG, SH, SZ, SO, TG, TI, UR, VD, VS, ZG, ZH
+     - None
    * - Taiwan
      - TW
      - None
+     - None
    * - Thailand
      - TH
      - None
+     - holidays[islamic]
```
Comment (KJhellico) [inline README.rst:543]:
```suggestion
     - None
```

File: README.rst [hunk 5]
```diff
@@ -113,376 +113,498 @@ for a subdivision its `ISO 3166-2 code`_. The following countries and
...
    * - Tunisia
      - TN
      - None
+     - None
```
Comment (KJhellico) [inline README.rst:547]:
```suggestion
     - holidays[islamic]
```

File: setup.cfg [hunk 6]
```diff
@@ -33,16 +33,30 @@ packages =
     holidays/countries
     holidays/financial
 install_requires =
-    PyMeeus
-    convertdate>=2.3.0
-    hijri-converter
-    korean-lunar-calendar
     python-dateutil
     backports.zoneinfo;python_version<'3.9'
     tzdata;os_name=='nt'
```
Comment (KJhellico) [inline setup.cfg:38]:
```suggestion
```

File: setup.cfg [hunk 7]
```diff
@@ -33,16 +33,30 @@ packages =
...
 python_requires = >=3.7
 include_package_data = True
 
+[options.extras_require]
+all =
+    PyMeeus
+    convertdate>=2.3.0
+    hijri-converter
+    korean-lunar-calendar
```
Comment (KJhellico) [inline setup.cfg:47]:
```suggestion
    korean-lunar-calendar
    backports.zoneinfo;python_version<'3.9'
    tzdata;os_name=='nt'    
```

File: setup.cfg [hunk 8]
```diff
@@ -33,16 +33,30 @@ packages =
...
+astronomic =
+    PyMeeus
```
Comment (KJhellico) [inline setup.cfg:49]:
```suggestion
    PyMeeus
    backports.zoneinfo;python_version<'3.9'
    tzdata;os_name=='nt'
```

File: setup.cfg (hunk 6 above)
Comment (arkid15r) [reply setup.cfg]: Done, 👍

**Review by KJhellico — COMMENTED** [review]
LGTM.

**Comment by arkid15r** [comment]
It seems that we've waited long enough to come up with an alternative solution.

I believe we will be able to get rid of the [all?] extra dependencies  using https://github.com/dr-prodigy/python-holidays/pull/1090 and https://github.com/dr-prodigy/python-holidays/pull/1094 approach.

**Comment by arkid15r** [comment]
This was a decent try. 

Closing this out as we have literally [one runtime dependency left](https://github.com/dr-prodigy/python-holidays/blob/4c691d998d1f31285d1a564e5ababcd3b00e973a/setup.cfg#L36).

Thanks to all participants!

**Pull request closed without merging**